### PR TITLE
added \n option

### PR DESCRIPTION
**Pull request by wanderer**
```
spaceing = 2
generate({
            kind: 'binop',
            type: 'i32',
            operator: 'add',
            left: {
                kind: 'get_local',
                id: {
                    kind: 'identifier',
                    name: 'x'
                }
            },
            right: {
                kind: 'get_local',
                id: {
                    kind: 'identifier',
                    name: 'y'
                }
            }
        }, spaceing))
```
will result in
`(i32.add\n  (get_local\n    $x\n  )\n  (get_local\n    $y\n  )\n)`    
with no spacing the result will be
`(i32.add (get_local $x)(get_local $y))`

### CODE REVIEW

File: bin/generator.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ function bodyGen (obj, kind) {
         res.push(parse('res += ' + compositeName[kind]));
     }
     if (objKinds.length > 0) {
-        res.push(parse(`indent += '  '`));
+        res.push(parse(`indent += spaceString`));
```
Comment (drom) [inline bin/generator.js]: @wanderer it is redundant code in case of `spaceString = ''`

File: bin/generator.js [hunk 2]
```diff
@@ -78,12 +78,24 @@ function bodyGen (obj, kind) {
 function funcObject (obj) {
     var res = esprima.parse(`
         'use strict';
-        var res, indent;
+        var res, indent, spaceNum, spaceString = '';
```
Comment (drom) [inline bin/generator.js]: @wanderer how about option in generator to insert configurable or hardcoded indent and see on the benchmark?

Comment (wanderer) [inline bin/generator.js]: is there a benchmark somewhere?

Comment (wanderer) [inline bin/generator.js]: ok added benchmark

**Pull request merged**